plan/task_profile.go

```diff
@@ -27,6 +27,7 @@ type taskProfile interface {
 	cost() float64
 	copy() taskProfile
 	plan() PhysicalPlan
+	finishTask(ctx context.Context, allocator *idAllocator) taskProfile
```

Comment:
The `finishTask` only make sense in `copTaskProfile`, promote it to `taskProfile` is confusing.